src/datasources/asset/AssetDataSource.ts

```diff
@@ -0,0 +1,325 @@
+import { DataFrameDTO, DataQueryRequest, DataSourceInstanceSettings, dateTime, MetricFindValue, TestDataSourceResponse, } from '@grafana/data';
+import { BackendSrv, getBackendSrv, getTemplateSrv, TemplateSrv } from '@grafana/runtime';
+import { DataSourceBase } from 'core/DataSourceBase';
+import {
+  AssetFilterProperties,
+  AssetModel, AssetQuery,
+  AssetQueryType,
+  AssetsResponse,
+  AssetUtilizationHistory,
+  AssetUtilizationHistoryResponse,
+  AssetUtilizationOrderBy,
+  AssetVariableQuery,
+  EntityType,
+  Interval,
+  IsNIAsset,
+  IsPeak,
+  PolicyOption,
+  QueryAssetUtilizationHistoryRequest,
+  ServicePolicyModel,
+  TimeFrequency,
+  UtilizationCategory,
+  Weekday
+} from './types';
+import {
+  calculateUtilization,
+  divideTimeRangeToBusinessIntervals,
+  extractTimestampsFromData,
+  filterDataByTimeRange,
+  groupDataByIntervals,
+  mergeOverlappingIntervals,
+  momentToTime,
+  patchMissingEndTimestamps,
+  patchZeroPoints
+} from "./helper";
+import { replaceVariables } from "../../core/utils";
+import { SystemMetadata } from "../system/types";
+import { defaultOrderBy, defaultProjection } from "../system/constants";
+import { isPeakLabels, timeFrequencyLabels } from "./constants";
+
+export class AssetDataSource extends DataSourceBase<AssetQuery> {
+  constructor(
+    readonly instanceSettings: DataSourceInstanceSettings,
+    readonly backendSrv: BackendSrv = getBackendSrv(),
+    readonly templateSrv: TemplateSrv = getTemplateSrv()
+  ) {
+    super(instanceSettings, backendSrv, templateSrv);
+  }
+
+  baseUrl = this.instanceSettings.url + '/niapm/v1';
+  sysmgmtUrl = this.instanceSettings.url + '/nisysmgmt/v1'
+
+  defaultQuery = {
+    assetQueryType: AssetQueryType.METADATA,
+    workspace: '',
+    entityType: EntityType.ASSET,
+    assetIdentifier: '',
+    isNIAsset: IsNIAsset.BOTH,
+    minionId: '',
+    isPeak: IsPeak.PEAK,
+    timeFrequency: TimeFrequency.DAILY,
+    utilizationCategory: UtilizationCategory.ALL,
+    peakDays: [Weekday.Monday, Weekday.Tuesday, Weekday.Wednesday, Weekday.Thursday, Weekday.Friday],
+    policyOption: PolicyOption.DEFAULT,
+    peakStart: undefined,
+    nonPeakStart: undefined
+  };
+
+  async runQuery(query: AssetQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
+    const result: DataFrameDTO = { refId: query.refId, fields: [] };
+
+    if (query.assetQueryType === AssetQueryType.METADATA) {
+      let workspace_id = this.templateSrv.replace(query.workspace);
+      let minion_id = this.templateSrv.replace(query.minionId);
```

Comment:
```suggestion
      let workspaceId = this.templateSrv.replace(query.workspace);
      let minionId = this.templateSrv.replace(query.minionId);
```

We always use camelCase in TS. This comment applies to a few other places.